File: app/src/main/java/com/platdmit/simplecloudmanager/domain/repo/AccountRepo.java
package com.platdmit.simplecloudmanager.domain.repo;

import com.platdmit.simplecloudmanager.domain.models.UserAccount;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

public interface AccountRepo {
    Single<UserAccount> getActiveAccount();
    Single<UserAccount> getPrepareAccountInfo(String login, String pass);
    Completable addAccountPin(UserAccount account);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/repo/AccountRepo.kt
package com.platdmit.simplecloudmanager.domain.repo

import com.platdmit.simplecloudmanager.domain.models.UserAccount
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface AccountRepo {
    fun getActiveAccount(): Single<UserAccount?>
    fun getPrepareAccountInfo(login: String, pass: String): Single<UserAccount?>
    fun addAccountPin(account: UserAccount): Completable
}